tests/Microsoft.Identity.Test.Unit/CoreTests/InstanceTests/GenericAuthorityTests.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Instance.Oidc;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.Identity.Test.Common.Core.Mocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.CoreTests.InstanceTests
+{
+    [TestClass]
+    public class GenericAuthorityTests : TestBase
+    {
+        [DataTestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        /// AAD doesn't returns the "scope" in the response
+        /// Duende does return the "scope" in the response
+        public async Task GenericClientCredentialSuccessfulTestAsync(bool includeScopeInResonse)
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                string authority = "https://demo.duendesoftware.com";
+                IConfidentialClientApplication app = ConfidentialClientApplicationBuilder
+                    .Create(TestConstants.ClientId)
+                    .WithHttpManager(httpManager)
+                    .WithExperimentalFeatures(true)
+                    .WithGenericAuthority(authority)
+                    .WithClientSecret(TestConstants.ClientSecret)
+                    .Build();
+
+                httpManager.AddMockHandler(
+                    CreateOidcHttpHandler(authority + @"/" + OidcRetrieverWithCache.OpenIdConfigurationEndpointSuffix));
+
+                httpManager.AddMockHandler(
+                    CreateTokenResponseHttpHandler(authority + "/connect/token", "api", includeScopeInResonse ? "api" : null));
+
+                Assert.AreEqual(authority + "/", app.Authority);
+                var confidentailClientApp = (ConfidentialClientApplication)app;
+                Assert.AreEqual(AuthorityType.Generic, confidentailClientApp.AuthorityInfo.AuthorityType);
+
+                AuthenticationResult result = await app
+                    .AcquireTokenForClient(new[] { "api" })
+                    .ExecuteAsync(CancellationToken.None)
+                    .ConfigureAwait(false);
+
+                Assert.IsNotNull(result);
+                Assert.IsNotNull(result.AccessToken);
+                Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+
+                result = await app
+                    .AcquireTokenForClient(new[] { "api" })
```

Comment:
Update the comment for [AcquireTokenForClient](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/g7ed6e%2Fgeneric-authority-support/src/client/Microsoft.Identity.Client/ConfidentialClientApplication.cs#L93) for scopes for generic authority. The scopes in the comment are specific to azure.